### PR TITLE
Migrate ZoneClient to SDK track v2

### PR DESCRIPTION
**Pull request by zarvd**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

Migrate ZoneClient to SDK track v2

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/provider/azure_zones.go
Comment (MartinForReal) [inline pkg/provider/azure_zones.go:1]: I'm wondering if these functions can be moved to repo package.

Comment (zarvd) [reply pkg/provider/azure_zones.go:1]: We could, but it’d be better to do it in a separate PR.

**Pull request merged**